### PR TITLE
[10.0] [IMP] hw_printer_network - Asynchroneous online/offline check …

### PR DESCRIPTION
**Pull request by thomaspaulb**
…of network devices (issue #513)

### CODE REVIEW

**Comment by thomaspaulb** [comment]
@GabbasovDinar @yelizariev Have finished the first version. Testing reveals a more consistently fast response from all three connected printers, although am still not trusting it to be fully bug / memory-leak-free. Would appreciate your critical review.

File: hw_printer_network/controllers/hw_printer_network_controller.py [hunk 1]
```diff
@@ -1,169 +1,178 @@
...
+                    return printer
+        if not found_printer:
+            self.add_network_printer(ip, name)
+        return None
+
+    def add_network_printer(self, ip, name=None):
+        printer = dict(
+            ip=ip,
+            status='offline',
+            # todo: start offline, then wait for online check
```
Comment (thomaspaulb) [inline hw_printer_network/controllers/hw_printer_network_controller.py]: this todo is actually done

**Comment by GabbasovDinar** [comment]
@thomaspaulb I will check on Sunday

File: hw_printer_network/controllers/hw_printer_network_controller.py [hunk 2]
```diff
@@ -1,169 +1,183 @@
...
-    from openerp.addons.hw_escpos.escpos.exceptions import NoDeviceError, HandleDeviceError, TicketNotPrinted, NoStatusError
-    from openerp.addons.hw_escpos.escpos.printer import Network
-    import openerp.addons.hw_escpos.controllers.main as hw_escpos_main
+    from odoo.addons.hw_escpos.escpos import escpos
+    from odoo.addons.hw_escpos.controllers.main import EscposProxy
+    from odoo.addons.hw_escpos.controllers.main import EscposDriver
+    from odoo.addons.hw_escpos.controllers.main import driver as usb_driver
+    from odoo.addons.hw_escpos.escpos.exceptions import NoDeviceError, HandleDeviceError, TicketNotPrinted, NoStatusError
+    from odoo.addons.hw_escpos.escpos.printer import Network
+    import odoo.addons.hw_proxy.controllers.main as hw_proxy
```
Comment (GabbasovDinar) [inline hw_printer_network/controllers/hw_printer_network_controller.py]: please, don't rename openerp to odoo, using openerp is necessary to support Raspberry Pi (the Odoo image has version 8.0)

File: pos_printer_network/static/src/js/network_printer.js [hunk 3]
```diff
@@ -134,29 +134,7 @@ odoo.define('pos_restaurant.network_printer', function (require) {
...
-                        }
-                    });
-                    return done;
-                };
-                connect = try_real_hard_to_connect(url,3);
-            }
+            connect = this._super(url, options);
             return connect.done(function(){
                 self.send_network_printers_to_pos_box(url, self.network_printers);
             });
```
Comment (GabbasovDinar) [inline pos_printer_network/static/src/js/network_printer.js]: I have a error " ReferenceError: connect is not defined".  
Use
```javascript
return this._super(url, options).done(function(){
    self.send_network_printers_to_pos_box(url, self.network_printers);
});
```

**Comment by GabbasovDinar** [comment]
@thomaspaulb Check please my comments and fix Travis errors

**Review by GabbasovDinar — APPROVED** [review]

File: hw_printer_network/controllers/hw_printer_network_controller.py (hunk 2 above)
Comment (thomaspaulb) [reply hw_printer_network/controllers/hw_printer_network_controller.py]: Hmm, have changed, but isn't it better to make a new updated image that contains Odoo 10.0 then?

**Comment by thomaspaulb** [comment]
@GabbasovDinar I think I fixed all. This needs some extensive testing in production though, that it still remains stable after some days of use.

**Comment by GabbasovDinar** [comment]
@thomaspaulb thank you for your great work :) You can add yourself to Contributors in README.rst

**Comment by thomaspaulb** [comment]
Thanks!

**Comment by GabbasovDinar** [comment]
@yelizariev we can merge it. After merge I update the documentation

**Comment by yelizariev** [comment]
Merged in #563 with few extra updates in docs. 
Thank you!

**Pull request closed without merging**